server-common/src/main/java/org/apache/kafka/timeline/SnapshotRegistry.java

```diff
@@ -205,6 +205,18 @@ public Snapshot getOrCreateSnapshot(long epoch) {
         return snapshot;
     }
 
+    /**
+     * Creates a new snapshot at the given epoch.
+     * <br>
+     * If {@code epoch} already exists, and it is the last snapshot then this operation will do nothing.
+     *
+     * @param epoch             The epoch to create the snapshot at. The current epoch
+     *                          will be advanced to one past this epoch.
+     */
+    public void idempotentCreateSnapshot(long epoch) {
+        getOrCreateSnapshot(epoch);
+    }
+
```

Comment:
This is the new method that calls the existing `getOrCreateSnapshot` but doesn't return anything, as it would leak the class outside its visibility scope.
I called it idempotent as this is what `getOrCreate` does effectively.